### PR TITLE
HOMS-415 Differentiate users from Keycloak and regular

### PR DESCRIPTION
**Pull request by theodor85**
Screenshot_20220512_140840

### CODE REVIEW

File: app/views/users/edit.html.haml [hunk 1]
```diff
@@ -4,5 +4,6 @@
                   method: 'put',
                   wrapper: :horizontal_form,
                   html: { class: 'form-horizontal order-edit-form col-sm-6' }) do |f|
-  = render partial: 'form_fields', locals: { f: f }
+  = render partial: 'form_fields', locals: { f: f, 
+                                             from_sso: @user.directory == 'hydra_sso' }
```
Comment (TimAle) [inline app/views/users/edit.html.haml:8]: I think `User` should have a check method like `from_sso?
Also, you write a symbol to this field but check a string.

File: db/schema.rb [hunk 2]
```diff
@@ -102,6 +102,9 @@
     t.string "password_salt"
     t.boolean "external", default: false, null: false
     t.boolean "blocked", default: false, null: false
+    t.string "provider"
+    t.string "uid"
```
Comment (TimAle) [inline db/schema.rb:106]: Where does it come from?

**Review by TimAle — CHANGES_REQUESTED** [review]

File: db/schema.rb (hunk 2 above)
Comment (theodor85) [reply db/schema.rb:106]: It's from `rake db:migrate`. I think it's old not applied changes

Comment (theodor85) [reply db/schema.rb:106]: It's seems to be some stuff from devise/omniauth: https://github.com/hydra-billing/homs/blob/f88f7264f2b74754277ed06e4b0f6279565fbb81/config/initializers/devise.rb#L234-L237

We don't use omniauth, so I'll remove it.

File: app/views/users/new.html.haml [hunk 3]
```diff
@@ -2,5 +2,6 @@
 .authform
   = simple_form_for @user || :user, url: add_users_path, html: { role: 'form'} do |f|
     %h3= t('.title')
-    = render partial: 'form_fields', locals: { f: f }
+    = render partial: 'form_fields', locals: { f: f, 
+                                             from_sso: @user.from_sso? }
```
Comment (TimAle) [inline app/views/users/new.html.haml:6]: alignment

**Review by TimAle — CHANGES_REQUESTED** [review]

**Review by TimAle — APPROVED** [review]

**Pull request merged**